rust-escrow/dao-factory/src/lib.rs

```diff
@@ -114,16 +124,37 @@ impl DaoFactory {
                     self.dao_index
                         .insert(&predecessor_account_id, &dao_account_id);
 
-                    return true;
+                    return self.create_ft(dao_name.clone());
                 }
 
-                Promise::new(predecessor_account_id).transfer(attached_deposit);
-
-                panic!("ERR_CREATE_DAO_UNSUCCESSFUL");
+                return Promise::new(predecessor_account_id).transfer(attached_deposit);
             }
-            _ => panic!("ERR_CREATE_DAO_UNSUCCESSFUL"),
+            _ => Promise::new(predecessor_account_id).transfer(attached_deposit),
         }
     }
+
+    #[payable]
+    fn create_ft(&mut self, name: String) -> Promise {
+        let account_id: AccountId = format!("{}-ft.{}", name, env::current_account_id())
+            .parse()
+            .unwrap();
+
+        let promise = Promise::new(account_id.clone())
+            .create_account()
+            .add_full_access_key(env::signer_account_pk())
+            .transfer(FT_BALANCE)
+            .deploy_contract(FT_CODE.to_vec());
+
+        promise
+            .function_call(
+                "new".to_string(),
+                json!({"owner_id": env::current_account_id(), "total_supply": FT_SUPPLY.to_string(), "metadata": { "spec": "ft-1.0.0", "name": account_id.to_string(), "symbol": "EXLT", "decimals": 8 }})
+                    .to_string()
+                    .into_bytes(),
+                0,
+                GAS_FOR_CREATE_FT,
+            )
+    }
```

Comment:
IMO, we should not mix the `DaoFactory` with the `FungibleTokenFactory`. Each factory deploys different standards. Let's follow the "separation of concerns principle".

What I suggest, is that we have 3 factories: 

`escrowfactory`

`daofactory`

`fungibletokenfactory`


Each doing what the do best and keeping a record of each type of contract they deploy.